src/main/java/com/amazonaws/services/sqs/AmazonSQSIdleQueueDeletingClient.java

```diff
@@ -189,17 +189,26 @@ public CreateQueueResult createQueue(CreateQueueRequest request) {
         if (!queueName.startsWith(queueNamePrefix)) {
             throw new IllegalArgumentException();
         }
-
+        String retentionPeriodString = retentionPeriod.get().toString();
+        int retry = 0;
         CreateQueueRequest superRequest = request.clone()
                 .withQueueName(queueName)
-                .withAttributes(attributes);
+                .withAttributes(attributes)
+                .addTagsEntry(IDLE_QUEUE_RETENTION_PERIOD_TAG, retentionPeriodString);
```

Comment:
Why don't we also include an initial heartbeat here too. That means that the queue will be protected during the initial random delay you're adding below.

Reply:
Included LAST_HEARTBEAT_TIMESTAMP_TAG in the `CreateQueueRequest`